pyreal/transformers/time_series_formatter.py

```diff
@@ -1,4 +1,480 @@
-from pyreal.transformers import Transformer
+"""
+The default and preferred format for multi-variable time-series data in Pyreal is
+a DataFrame with two levels of columns. The first level of columns represents the
+variables of the data, and the second level represents the number of timesteps in
+the signal(s). The Index of this DataFrame numerates the number of instances in
+the data.
+
+Let `data` be a time-series dataset in Pyreal format with 100 instances, 3 variables
+and 200 steps. The indexing schemes of Pyreal time-series format work similarly
+as Pandas DataFrame.
+
+Examples
+-------------------
+Access the first instance sequence of data
+>>> data.iloc[0]
+Access the value of variable "dim_0" at step 50
+>>> data["dim_0"][50]
+or
+>>> data["dim_0", 50]
+
+The indexing on instances and variables can also be used together
+>>> data["dim_0", 50].iloc[0]
+"""
+import numpy as np
+import pandas as pd
+
+from pyreal.transformers import Transformer, fit_transformers, run_transformers
+
+
+def _check_is_np2d(x):
+    if not isinstance(x, np.ndarray):
+        raise TypeError(f"Input data must be an np.ndarray, but found: {type(x)}")
+    if x.ndim != 2:
+        raise ValueError(f"Input data must have two dimensions, but found shape: {x.shape}")
+
+
+def _check_is_np3d(x):
+    if not isinstance(x, np.ndarray):
+        raise TypeError(f"Input data must be an np.ndarray, but found: {type(x)}")
+    if x.ndim != 3:
+        raise ValueError(f"Input data must have three dimensions, but found shape: {x.shape}")
+
+
+def _check_is_pd2d(x):
+    if not isinstance(x, pd.DataFrame):
+        raise TypeError(f"Input data must be a pd.DataFrame, but found: {type(x)}")
+    if x.ndim != 2:
+        raise ValueError(f"Input data must have two dimensions, but found shape: {x.shape}")
+
+
+def _check_is_pyreal_df(x):
+    if not is_valid_dataframe(x):
+        raise TypeError(
+            f"Input data must be a MultiIndex-ed column DataFrame, but found: {type(x)}"
+        )
+
+
+def _check_is_sktime_nest(x):
+    if not isinstance(x, pd.DataFrame):
+        raise TypeError(f"Input data must be a pd.DataFrame, but found: {type(x)}")
+    else:
+        if not x.applymap(lambda cell: isinstance(cell, pd.Series)).values.any():
+            raise ValueError("Entries of input data must be pd.Series")
+
+
+def is_valid_dataframe(x):
+    """
+    Check if the input is a valid DataFrame.
+
+    Args:
+        x: Input data
+    Returns:
+        bool: Whether the input is a MultiIndex DataFrame
+    """
+    if not isinstance(x, pd.DataFrame):
+        return False
+    elif not isinstance(x.columns, pd.MultiIndex):
+        return False
+    elif x.columns.nlevels != 2:
+        return False
+    return True
+
+
+class np2d_to_df(Transformer):
```

Comment:
Please change class names to adhere to the typical format (capital camelcase, ie, Np2dToDf

Reply:
I replaced all `df` with `MultiIndexFrame`.